### PR TITLE
Section 4 comments

### LINKED ISSUE #217: Section 4.1.1
**Organization / Program:** Slandala <br>**Section:** 4.1.1 <br>**PDF Page:** None<br>**PDF Line(s):** 699<br>**Comment:** Should be restrict requests to "authorized" or "approved" requestors? Or can anyone in an agency request a cert?<br><br>**Suggested Change:** None

### LINKED ISSUE #218: Section 4.3.1
**Organization / Program:** Slandala <br>**Section:** 4.3.1 <br>**PDF Page:** None<br>**PDF Line(s):** 761<br>**Comment:** I thought administrators could not issue certs?<br><br>**Suggested Change:** None

### LINKED ISSUE #220: Section 4
**Organization / Program:** Slandala <br>**Section:** 4 <br>**PDF Page:** None<br>**PDF Line(s):** 790<br>**Comment:** Is there any additional requirement to notify a PMA, other cross certified CAs, etc.<br><br>**Suggested Change:** None

### LINKED ISSUE #224: Section 4.9.5
**Organization / Program:** Slandala <br>**Section:** 4.9.5 <br>**PDF Page:** None<br>**PDF Line(s):** 950<br>**Comment:** The CP should identify a time period in which a validated request must be fulfilled. (often before the next CRL, unless within 2 hours)<br><br>**Suggested Change:** None

### LINKED ISSUE #225: Section 4.9.3
**Organization / Program:** Slandala <br>**Section:** 4.9.3 <br>**PDF Page:** None<br>**PDF Line(s):** 937<br>**Comment:** Is there a requirement to use reason codes, especially in the case of compromise.<br><br>**Suggested Change:** None

### LINKED ISSUE #252: Section 4.9.1.1
**Organization / Program:** Scott S. Perry CPA, PLLC <br>**Section:** 4.9.1.1 <br>**PDF Page:** 40<br>**PDF Line(s):** 872<br>**Comment:** The CP is silent on external government, judicial or regulatory authority to revoke a certificate<br><br>**Suggested Change:** ADD: The Issuer CA received a lawful and binding order from a government, judicial or regulatory body to 
revoke the Certificate;

### LINKED ISSUE #253: Section 4.9.1.2
**Organization / Program:** Scott S. Perry CPA, PLLC <br>**Section:** 4.9.1.2 <br>**PDF Page:** 41<br>**PDF Line(s):** 908<br>**Comment:** The CP is silent on external government, judicial or regulatory authority to revoke a certificate<br><br>**Suggested Change:** ADD: The Issuer CA received a lawful and binding order from a government, judicial or regulatory body to 
revoke the Certificate;

### LINKED ISSUE #283: Section 4.2.1
**Organization / Program:** FPKIMA <br>**Section:** 4.2.1 <br>**PDF Page:** 35<br>**PDF Line(s):** 741<br>**Comment:** 825 days is a long time in Government.  <br><br>**Suggested Change:** Consider shortening to 18 months, and include a requirement to validate the information is still correct prior to issuing.

### LINKED ISSUE #284: Section 4.2.1
**Organization / Program:** FPKIMA <br>**Section:** 4.2.1 <br>**PDF Page:** 35<br>**PDF Line(s):** 742-744<br>**Comment:** High Risk Certificates are not identified.  These lines just state that the issuing CA has to identify which ones are considered high risk, but that could result in inconsistent definitions of High Risk.<br><br>**Suggested Change:** Define High Risk certificates.  At a minimum, consider wildcard certificates as high risk.

### LINKED ISSUE #285: Section 4.3.1
**Organization / Program:** FPKIMA <br>**Section:** 4.3.1 <br>**PDF Page:** 36<br>**PDF Line(s):** 771-772<br>**Comment:** It's not appropriate in a Government document for the Government to specify by policy that the Government must do business with a specific company (i.e., Google).  If there is some requirement to use Google because of the Google Trust Store for Chrome, then Google will still end  up as one of two CT Logs.<br><br>**Suggested Change:** Replace these two lines with a statement along the lines of:

CT Logs must be operated by separate organizations in order to meet the requirements for publishing to CT Logs.

### LINKED ISSUE #286: Section 4.4.3
**Organization / Program:** FPKIMA <br>**Section:** 4.4.3 <br>**PDF Page:** 37<br>**PDF Line(s):** 790<br>**Comment:** The reference to Section 4.4.2 for this section does not address the requirement.<br><br>**Suggested Change:** Restate to:

The FPKIPA must be notified whenever a CA operating under this policy issues a CA certificate. Subscriber certificates issued via other than automated means will be notified using Reliable Methods of Communication.

### LINKED ISSUE #291: Section 4.10.2
**Organization / Program:** FPKIMA <br>**Section:** 4.10.2 <br>**PDF Page:** 44<br>**PDF Line(s):** 1021-1028<br>**Comment:**  RFC3647 states information on "any applicable policies on unavailablity" should be included.  Some mention of downtime for maintenance should probably be mentioned.<br><br>**Suggested Change:** Change language to: 

The publicly accessible repository system, including certificate status server, shall be designed and implemented so as to provide 99% availability overall, and limit scheduled downtime to 0.5% annually.

### LINKED ISSUE #309: Section 4.9.12
**Organization / Program:** FPKIMA <br>**Section:** 4.9.12 <br>**PDF Page:** 43-44<br>**PDF Line(s):** 1006-1008<br>**Comment:** Special Requirements for re key compromise just references a section that states under what conditions a certificate will be revoked.  This section should include requirements to notify the FPKI PA of suspected compromise, requirements to allow access to appropriate law enforcement organizations for investigation, qualifications for investigators (or potentially state that funding the investigation is the responsibility of the CA.)<br><br>**Suggested Change:** In the case of a compromise of the private key used to sign certificates, the CA must immediately notify the FPKI PA that the CA certificate has been compromised. The Issuing CA will revoke the signing key, and publish a CRL to make relying parties aware that the certificates off this signing key can no longer be trusted.
The CA is responsible for investigating the circumstances of any such compromise with individuals qualified to perform computer forensic investigations, must provide results of the investigation to the FPKIPA, as well as work with appropriate law enforcement organizations.

### LINKED ISSUE #311: Section 4.9.7
**Organization / Program:** FPKIMA <br>**Section:** 4.9.7 <br>**PDF Page:** 42<br>**PDF Line(s):** 966<br>**Comment:** Missing Root CA CRL issuance frequency<br><br>**Suggested Change:** add it as part of Sub CA

### LINKED ISSUE #352: Section 4.1.2
**Organization / Program:** DoD <br>**Section:** 4.1.2 <br>**PDF Page:** 34<br>**PDF Line(s):** 708<br>**Comment:** References a "Subscriber Agreement or Terms of Use" but does not provide any information about what this document is or what is required to be contained in it.<br><br>**Suggested Change:** Use one term for this role and apply it consistently throughout the document (e.g., Sections 1.3, Section 5.2.1, Section 5.3.3) for clarity

### LINKED ISSUE #353: Section 4.1.2
**Organization / Program:** DoD <br>**Section:** 4.1.2 <br>**PDF Page:** 35<br>**PDF Line(s):** 722<br>**Comment:** Which public/private key pair?  Issuing CA or requested certificate?<br><br>**Suggested Change:** Clarify

### LINKED ISSUE #354: Section 4.2.1
**Organization / Program:** DoD <br>**Section:** 4.2.1 <br>**PDF Page:** 35<br>**PDF Line(s):** 736-737<br>**Comment:** Do the exceptions cited in 3.2.2.8 apply?<br><br>**Suggested Change:** Be explicit by either saying 3.2.2.8 exceptions do not apply or apply

### LINKED ISSUE #357: Section 4.2.2
**Organization / Program:** DoD <br>**Section:** 4.2.2 <br>**PDF Page:** 35<br>**PDF Line(s):** 747<br>**Comment:** A certificate policy is a document, it can't be "technically constrained"<br><br>**Suggested Change:** Consider using the term "security auditor" for the role described in section 5, and "compliance auditor" for the role described in section 8. Regardless, section 5 needs to use consistent terminology.

### LINKED ISSUE #360: Section 4.4.2
**Organization / Program:** NSA <br>**Section:** 4.4.2 <br>**PDF Page:** 37<br>**PDF Line(s):** 786-787<br>**Comment:** Section 2.3.1 states that a CA cert has to be posted w/in 15 days.  It is unclear why the CT requirement needs to be listed here as well as Section 4.3.1.<br><br>**Suggested Change:** reference Section 2.3.1 for CA certificates.  And remove the
 requirements statement for end entity certificates.

### LINKED ISSUE #362: Section 4.9.1 (numbered bullet 13)
**Organization / Program:** DoD <br>**Section:** 4.9.1 (numbered bullet 13) <br>**PDF Page:** 40<br>**PDF Line(s):** 897-898<br>**Comment:** This is poorly written since it indicates use of indirect CRL which the profiles do not support.  The CA can revoke a sub CA but not certificate issued by it.  A CA can direct the sub CA to revoke a certificate, but that is hollow and no security.  Sub CA key is either compromised or not.  If it is not, there is useless effort.  If it is, even asking Sub CA to revoke a certificate is not useful since it creates a race condition with the adversary and false sense of security.  Only practical thing to do is revoke the sub CA.<br><br>**Suggested Change:** delete this bullet

### LINKED ISSUE #363: Section 4.9.2
**Organization / Program:** DoD <br>**Section:** 4.9.2 <br>**PDF Page:** 41<br>**PDF Line(s):** 936<br>**Comment:** Section 4.9.1.2 requires a CA to revoke a subordinate CA certificate if any of the listed conditions occurs, but this section states that the FPKIPA has to direct revocation of a CA certificate. These are conflicting requirements<br><br>**Suggested Change:** Add details for what the results of each area should be, such as:
* A criminal history check shall show no misdemeanor or felony conviction;
* A credit history check shall show that person has not committed any fraud or is otherwise financially trustworthy;
* Employment verification shall demonstrate that the person is competent, reliable and trustworthy;
* Professional references shall demonstrate that the person is competent, reliable and trustworthy;
* Education verification of highest or most relevant degree

### LINKED ISSUE #364: Section 4.9.6
**Organization / Program:** DoD <br>**Section:** 4.9.6 <br>**PDF Page:** 42<br>**PDF Line(s):** 962-965<br>**Comment:** With the recommended text in Section 4.5.2, point to 4.5.2<br><br>**Suggested Change:** Replace the text with See Section 4.5.2

### LINKED ISSUE #370: Section 4.9.10
**Organization / Program:** DoD <br>**Section:** 4.9.10 <br>**PDF Page:** 43<br>**PDF Line(s):** 993-994<br>**Comment:** Need to put onus on the Responder and not the CA just like the prior bullet.  35 days is selected assuming other suggestion is taken.<br><br>**Suggested Change:** Replace with "The OCSP Responder SHALL update the information used to respond to requests within 4 hours of a new CRL being issued by the CA for all requests. OCSP responses from this service SHALL have a maximum expiration time of 35 days"

### LINKED ISSUE #371: Section 4.9.10 - OCSP responses for not Technically Constrained
**Organization / Program:** DoD <br>**Section:** 4.9.10 <br>**PDF Page:** 43<br>**PDF Line(s):** 998-999<br>**Comment:** Is this for the status of the CA certificates or for the status of certificates issued by such CAs or both?<br><br>**Suggested Change:** Clarify.  I would think it is for both.

### LINKED ISSUE #372: Section 4.9.12
**Organization / Program:** DoD <br>**Section:** 4.9.12 <br>**PDF Page:** 44<br>**PDF Line(s):** 1007-1008<br>**Comment:** 24 hour requirement contradicts 4.9.1.2 in general as well as specifically for key compromise (see bullet 3 in 4.9.1.2)<br><br>**Suggested Change:** Reconcile to make the text consistent.  In addition, in 4.9.12 text (over and above title of the section) use the term "key compromise" explicitly.

### PR DESCRIPTION
**Pull request by lachellel**
57 comments reviewed
42 addressed with changes or no changes

Outstanding items: 
1.  lifetime and frequency for CRLs and OCSP - updates made as placeholders only; final will be discussed
2.  validity lifetime of identity evidence (#72)
3.  delegated ocsp responder certificates and renewals, and ocsp signing and responses in general

closes #217 closes #352 closes #353 closes #291 closes #283 closes #284 closes #354 closes #357 closes #285 closes #218 closes @219 closes #360 closes #286 partial #467 closes #362 closes #252 closes #253 closes #370 closes #371 closes #309 closes #372 closes #363 closes #225 closes #224 closes #364 closes #311 closes #220

### CODE REVIEW

**Comment by lachellel** [comment]
@konklone 

I just opened #488 - a PR from section5-comments branch to section4-comments branch.  Suggest we merge / accept that PR and all changes will flow to this branch and PR.

**Comment by lachellel** [comment]
For section 6 updates, I'll edit against a copy of master so it's not out of sync!

File: certificate-policy.md [hunk 1]
```diff
@@ -434,82 +426,77 @@ All CAs shall specify the procedures for validating information and identity evi
...
 
-The Issuing CA shall submit a precertificate to a minimum of TWO Certificate Transparency Logs for certificates with a validity period less than or equal to 395 days.  The Issuing CA shall submit a precertificate to a minimum of THREE Certificate Transparency Logs for certificates with a validity period greater than 395 days and less than or equal to 825 days. There is no limit on the maximum number of CT Logs which may be submitted to.
-
-- At least one of the Certificate Transparency Logs shall be a CT Log operated by Google.
-- At least one of the Certificate Transparenty Logs shall be a CT Log not operated by Google.
+- At least one of the CT Logs shall be a log operated by Google.
+- At least one of the CT Logs shall be a Log operated by a government or business entity other than Google.
```
Comment (konklone) [inline certificate-policy.md]: Nit: inconsistent capitalization of `log`/`Log` in these two lines.

File: certificate-policy.md [hunk 2]
```diff
@@ -669,26 +656,22 @@ In the latter case, the OCSP signing Certificate shall contain an extension of t
...
 
-OCSP responders for CAs which are not Technically Constrained in line with Section 7.1.5 shall not respond with a "good" status for such certificates.
-
 ### 4.9.11 Other forms of revocation advertisements available
-If the Subscriber Certificate is for a high-traffic FQDN, the CA may rely on stapling, in accordance with [RFC4366], to distribute its OCSP responses. In this case, the CA shall ensure that the Subscriber "staples" the OCSP response for the Certificate in its TLS handshake. The CA shall enforce this requirement on the Subscriber either contractually, through the Subscriber Agreement or Terms of Use, or by technical review measures implemented by the CA.
+If the Subscriber Certificate is for a high-traffic FQDN, the CA may rely on stapling, in accordance with RFC4366, to distribute its OCSP responses. In this case, the CA shall ensure that the Subscriber "staples" the OCSP response for the Certificate in its TLS handshake. The CA shall enforce this requirement on the Subscriber either contractually, through the Subscriber Agreement or Terms of Use, or by technical review measures implemented by the CA.
```
Comment (konklone) [inline certificate-policy.md:669]: Is this compatible with root store requirements, that you could exempt OCSP responses being available by `GET` requests if you can ensure stapling?

**Review by konklone — APPROVED** [review]
I added a nit comment, and asked a question. Neither should block the merge, so unless the question points to a significant issue, merge away. :+1:

File: certificate-policy.md (hunk 2 above)
Comment (lachellel) [reply certificate-policy.md:669]: @konklone The comment isn't a "nit"; your comment is valid.  

This policy statement is directly from the Baseline Requirements.  A related comment is in Issue #290 to "define a high-traffic FQDN".  I added your comments and mine to Issue #290.

Comment (konklone) [reply certificate-policy.md:669]: Thanks. And it was [my other comment](https://github.com/uspki/policies/pull/484#discussion_r158415034) that was the nit. :)

**Pull request merged**